Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/Limits.h

```diff
@@ -96,8 +96,8 @@ namespace AZ::RHI
         constexpr DeviceMask AllDevices{ static_cast<DeviceMask>(AZStd::numeric_limits<uint32_t>::max()) };
         constexpr DeviceMask DefaultDevice{ 1u };
 
-        constexpr uint32_t DefaultDeviceIndex { 0 };
+        constexpr int DefaultDeviceIndex{ 0 };
```

Comment:
any reason for switching to int? Can this just be uint16_t?

Reply:
We are trying to avoid unsigned data types if really not necessary. I would actually even make the `DeviceMask` signed. There are several sources already available on the web as to why signed should be preferred over unsigned (e.g. https://blog.libtorrent.org/2016/05/unsigned-integers/ or for a more specific part, but from Bjarne Stroustrup himself: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1428r0.pdf). Also why would you use 16 bit? given that the mask currently supports at maximum 32 devices 8 bit would be enough, but I don't see a reason to go below 32 bit since that is the smallest machine word size of all supported architectures, if I'm not wrong?